### PR TITLE
Restructuring tuned profile hierarchy.

### PR DESCRIPTION
**Pull request by jmencak**
Proposing openshift-{control-plane,node} profiles.  The plan is to eventually retire origin-node-{guest,host} profiles.

The current profile structure:

    throughput-performance
        |          |______________
        |                         |
    virtual-guest                 |
        |                         |
        |                         |
    origin-node-guest       origin-node-host

Proposed profile structure:

    throughput-performance
                 |
                 |
            openshift (using tuned 2.8.x auto-parent functionality)
                 |
        ---------+---------------
       |                         |
       |                         |
    openshift-control-plane   openshift-node

There is an associated  https://github.com/openshift/openshift-ansible/pull/4566 against openshift-ansible.  This PR adds a dependency on tuned 2.8.x, because of its auto-parent functionality.

### CODE REVIEW

File: contrib/tuned/openshift-control-plane/tuned.conf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# default is 10%)
+vm.dirty_background_ratio=3
+
+# The swappiness parameter controls the tendency of the kernel to move
+# processes out of physical memory and onto the swap disk.
+# 0 tells the kernel to avoid swapping processes out of physical memory
+# for as long as possible
+# 100 tells the kernel to aggressively swap processes out of physical memory
+# and move them to swap cache
+vm.swappiness=10
```
Comment (jeremyeder) [inline contrib/tuned/openshift-control-plane/tuned.conf]: how/why is this here? I think you copy/pasted from latency-performance?

Comment (jeremyeder) [inline contrib/tuned/openshift-control-plane/tuned.conf]: specifically referring to the vm.* tunables.

**Comment by jeremyeder** [comment]
tuned profile content LGTM.  Thanks for the update @jmencak! 

This will require tuned from RHEL7.4 and have the effect of making tuned require RHEL7.4 content (only tuned RPM) be available in order to install OpenShift 3.7.

@sdodson @brenton any concerns on that specifically?

**Comment by brenton** [comment]
The logic makes sense to me.  We'd need to wait to merge this until after we've branched for 3.6.

File: contrib/tuned/openshift-control-plane/tuned.conf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#
+# tuned configuration
+#
+
+[main]
+summary=Optimize systems running OpenShift control plane
+include=openshift
+
+[sysctl]
+# ktune sysctl settings for rhel6 servers, maximizing i/o throughput
```
Comment (sdodson) [inline contrib/tuned/openshift-control-plane/tuned.conf]: s/rhel6 servers,// ?

File: origin.spec [hunk 3]
```diff
@@ -567,22 +574,10 @@ fi
...
-# reset the tuned profile to the recommended profile
-# $1 = 0 when we're being removed > 0 during upgrades
-if [ "$1" = 0 ]; then
-  recommended=`/usr/sbin/tuned-adm recommend`
-  /usr/sbin/tuned-adm profile $recommended > /dev/null 2>&1
-fi
+%{_prefix}/lib/tuned/openshift
+%{_prefix}/lib/tuned/openshift-control-plane
+%{_prefix}/lib/tuned/openshift-node
+%{_sysconfdir}/tuned/recommend.conf
```
Comment (sdodson) [inline origin.spec]: There's no scripts necessary to trigger profile change?

These are being included in the `tuned-profiles-%{name}-node` package. Should that be renamed to be more generic? If so we should provide obsoletes and we should adjust the requires for master and node subpackages to require the new package. I think we can get by with it as is because all masters should be nodes. Will masters get both policies unioned?

**Comment by sdodson** [comment]
I'm fine with that timing.

File: origin.spec (hunk 3 above)
Comment (jmencak) [inline origin.spec]: The profile changes automatically to the recommended profile once the "tuned" daemon is restarted by the ansible installer (see https://github.com/openshift/openshift-ansible/pull/4566).

Yes, I was thinking of using something like tuned-profiles-openshift instead of tuned-profiles-%{name}-node for both an openshift master and nodes.  Would that be OK?  Not sure all masters should be nodes.  Are you sure about that?  I've installed OpenShift with a master without being a "node", meaning, I'd have to adjust https://github.com/openshift/openshift-ansible/pull/4566 too.  That's why I've created an ansible role for tuned, to be included not only from nodes, but masters too eventually.

**Comment by jmencak** [comment]
Closing in favour of https://github.com/openshift/openshift-ansible/pull/5054 and https://github.com/openshift/openshift-ansible/pull/5089
